src/components/shared/Search/SearchContext.tsx

```diff
@@ -0,0 +1,65 @@
+import React, { Component } from 'react'
+import { Docs, DocsRenderProps } from 'docz'
+import { getMenusFromDocs, Menus } from '../../../utils/getMenusFromDocs'
+
+interface Props {
+  docs: DocsRenderProps['docs']
+}
+
+export interface SearchContext {
+  menus: Menus | []
+  searching: boolean
+  setSearching: (arg0: boolean) => void
+  setMenus: (arg0: Menus) => void
+}
+
+type State = SearchContext
+
+const SearchContext = React.createContext<State | null>(null)
+
+class Provider extends Component<Props, State> {
+  constructor(props: Props) {
+    super(props)
+    this.state = {
+      menus: [],
+      searching: false,
+      setMenus: this.setMenus,
+      setSearching: this.setSearching,
+    }
+  }
+
+  public componentDidMount(): void {
+    const initial = getMenusFromDocs(this.props.docs)
```

Comment:
Why set these here and not the constructor?

Reply:
Fixed.